sources/packages/backend/apps/api/src/route-controllers/program-year/_tests_/e2e/program-year.students.controller.getProgramYearById.e2e-spec.ts

```diff
@@ -0,0 +1,120 @@
+      programYear: "2001-2002",
+      programYearDesc:
+        "Study starting between August 01, 2001 and July 31, 2002",
+      formName: "SFAA2001-02",
+      active: false,
+      parentFormName: "SFAA2001-01-parent",
+      partnerFormName: "SFAA2001-01-partner",
+      programYearPrefix: "2001",
+    } as ProgramYear;
+
+    programYear2026 = {
+      programYear: "2026-2027",
+      programYearDesc:
+        "Study starting between August 01, 2026 and July 31, 2027",
+      formName: "SFAA2026-27",
+      active: true,
+      parentFormName: "SFAA2026-01-parent",
+      partnerFormName: "SFAA2026-01-partner",
+      programYearPrefix: "2026",
+    } as ProgramYear;
+
+    const programYear2001Promise = programYearRepo.save(programYear2001);
+    const programYear2026Promise = programYearRepo.save(programYear2026);
+
+    [programYear2001, programYear2026] = await Promise.all([
+      programYear2001Promise,
+      programYear2026Promise,
+    ]);
+
+    app = nestApplication;
+  });
+
+  it("Should get an active program year by id when available.", async () => {
+    // Arrange
+    const endpoint = `/students/program-year/${programYear2026.id}/active`;
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        id: programYear2026.id,
+        formName: programYear2026.formName,
+        programYear: programYear2026.programYear,
+        programYearDesc: programYear2026.programYearDesc,
+      });
+  });
+
+  it("Should return 'not found' status when a program year does not exist.", async () => {
+    // Arrange
+    const endpoint = "/students/program-year/9999/active";
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.NOT_FOUND)
+      .expect({
+        statusCode: 404,
+        message: "Program Year Id 9999 was not found.",
+        error: "Not Found",
+      });
+  });
+
+  it("Should return 'not found' status when a program year is inactive.", async () => {
+    // Arrange
+    const endpoint = `/students/program-year/${programYear2001.id}/active`;
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.NOT_FOUND)
+      .expect({
+        statusCode: 404,
+        message: `Program Year Id ${programYear2001.id} was not found.`,
+        error: "Not Found",
+      });
+  });
+
+  afterAll(async () => {
+    const programYearRepo = appDataSource.getRepository(ProgramYear);
```

Comment:
As a general rule, we have not been cleaning the data after the tests are executed.